technique/muon/tektronix_osc_screenshots.py

```diff
@@ -0,0 +1,91 @@
+import os
+import requests
+import sys
+import shutil
+
+from datetime import datetime
+from io import BytesIO
+from PIL import Image
+from time import sleep
+from zipfile import ZipFile
+
+sys.path.insert(0, os.path.abspath(os.path.join(r"C:\\", "Instrument", "scripts")))
+sys.path.insert(0, os.path.abspath(os.path.join(os.environ["KIT_ROOT"], "ISIS", "inst_servers", "master")))
+
+from genie_python import genie as g
+from genie_python.channel_access_exceptions import UnableToConnectToPVException
+from server_common.helpers import register_ioc_start
+
+register_ioc_start("BGRSCRPT_03")
+
+g.set_instrument(None)
+
+# User editable constants
+DEVICE_IP = "130.246.50.7"
+INSTRUMENT = "EXAMPLE_INST"
+MINS_BETWEEN_SCREENSHOTS = 10
+
+TESTING = False  # Stops the permanent while loop running for testing the functions. 
+
+def get_filename(rb_number: str) -> str:
+    return rb_number + "_" + datetime.now().replace(microsecond=0).isoformat().replace(":", "-") + ".png"
+
+def get_image() -> Image:
+    response = requests.get(f"http://{DEVICE_IP}/image.png", timeout=10)
+    return Image.open(BytesIO(response.content))
+
+while not TESTING:
+    try:
+        print("Creating filenames...")
+        run_number = g.get_runnumber()
+        rb_number = g.get_rb()
+        base_dir = "C:\\data\\"
+        zip_base_file_path = base_dir + "NDX" + INSTRUMENT + run_number
+        zip_archive_file_path = zip_base_file_path + "_scope_screens.zip"
+        zip_temp_file_path = zip_base_file_path + "_scope_screens_temp.zip"
```

Comment:
probably safer to do this using os.path.join and fstrings. something like this perhaps?
```suggestion
        base_dir = os.path.join("C:", "data")
        zip_base_file_path =  f"NDX{INSTRUMENT}{run_number}"
        zip_archive_file_path = os.path.join(base_dir, f"{zip_base_file_path}_scope_screens.zip")
        zip_temp_file_path =  os.path.join(base_dir,f"{zip_base_file_path}_scope_screens_temp.zip")
```

Reply:
This one will require some fiddling around with, to get the script working again.